packages/wonder-blocks-data/src/hooks/use-cached-effect.js

```diff
@@ -104,107 +120,152 @@ export const useCachedEffect = <TData: ValidCacheData>(
         // that all calls when the request is in-flight will update once that
         // request is done, we want the cache to be empty until that point.
     );
-
-    // Build a function that will update the cache and either invoke the
-    // callback provided in options, or force an update.
     const forceUpdate = useForceUpdate();
-    const setCacheAndNotify = React.useCallback(
-        (value: Result<TData>) => {
-            setMostRecentResult(value);
-
-            // If our caller provided a cacheUpdated callback, we use that.
-            // Otherwise, we toggle our little state update.
-            if (onResultChanged != null) {
-                onResultChanged(value);
-            } else {
-                forceUpdate();
+    // For the NetworkOnly fetch policy, we ignore the cached value.
+    // So we need somewhere else to store the network value.
+    const networkResultRef = React.useRef();
```

Comment:
For the NetworkOnly fetch policy, we need a way to track that network result outside of the shared cache. I did try using state and replacing the force update but I was getting some infinite render loop issues and I couldn't quite track it down. So for now, I'll do it this way where the value is stored as a ref and we just look at it on render to determine what to do.